### PR TITLE
Mon-148838 Add a robot test to check the service 

### PR DESCRIPTION
**Pull request by sechkem**
Robot test to validate the behavior of the old engine configuration service inheritance in the new one.

REFS: MON-148838

### CODE REVIEW

File: common/engine_legacy_conf/service.hh [hunk 1]
```diff
@@ -190,7 +190,7 @@ class service : public object {
   std::string _host_name;
   std::string _icon_image;
   std::string _icon_image_alt;
-  uint32_t _initial_state;
+  opt<unsigned int> _initial_state;
```
Comment (bouda1) [inline common/engine_legacy_conf/service.hh:193]: are you sure we have to make this field inheritable?

File: tests/resources/Engine.py [hunk 2]
```diff
@@ -869,18 +869,55 @@ def ctn_engine_config_set_value_in_services(idx: int, desc: str, key: str, value
         key (str): The key whose value needs to change.
         value (str): The new value to set.
     """
-    filename = f"{ETC_ROOT}/centreon-engine/config{idx}/services.cfg"
+    filename = f"{ETC_ROOT}/centreon-engine/config{idx}/{file}"
     with open(filename, "r") as f:
         lines = f.readlines()
+        
+    if file =="serviceTemplates.cfg":
```
Comment (bouda1) [inline tests/resources/Engine.py:876]:
```suggestion
    if file == "serviceTemplates.cfg":
```

File: tests/resources/Engine.py [hunk 3]
```diff
@@ -869,18 +869,55 @@ def ctn_engine_config_set_value_in_services(idx: int, desc: str, key: str, value
...
+        key (str): the parameter that will be deleted.
+        file (str): The file to modify, default value 'service.cfg'
+    """
+    
+
+    filename = f"{ETC_ROOT}/centreon-engine/config{idx}/{file}"
+    with open(filename, "r") as f:
+        lines = f.readlines()
+
+    if file =="serviceTemplates.cfg":
```
Comment (bouda1) [inline tests/resources/Engine.py:904]:
```suggestion
    if file == "serviceTemplates.cfg":
```

File: tests/resources/Engine.py [hunk 4]
```diff
@@ -3757,6 +3794,33 @@ def ctn_get_host_info_grpc(id:int):
...
+    Retrieve service information via a gRPC call.
+
+    Args:
+        id_h: The identifier of the host to retrieve.
+        id_s: The identifier of the service to retrieve.
+
+    Returns:
+        A dictionary containing the service informations, if successfully retrieved.
+    """
+    if id_h is not None and id_s is not None:
```
Comment (bouda1) [inline tests/resources/Engine.py:3808]: As id_h and id_s are declared as int, is this test relevant?

File: common/engine_legacy_conf/service.cc [hunk 5]
```diff
@@ -705,6 +705,7 @@ void service::merge(object const& obj) {
   service const& tmpl(static_cast<service const&>(obj));
 
   MRG_OPTION(_acknowledgement_timeout);
+  MRG_OPTION(_initial_state);
```
Comment (bouda1) [inline common/engine_legacy_conf/service.cc:708]: are you sure we have to make this field inheritable?

File: tests/resources/Engine.py (hunk 2 above)
Comment (sechkem) [reply tests/resources/Engine.py:876]: will fix

File: tests/resources/Engine.py (hunk 3 above)
Comment (sechkem) [reply tests/resources/Engine.py:904]: will fix

File: tests/resources/Engine.py (hunk 4 above)
Comment (sechkem) [reply tests/resources/Engine.py:3808]: will fix, the test isn't necessary.

File: common/engine_legacy_conf/service.hh (hunk 1 above)
Comment (sechkem) [reply common/engine_legacy_conf/service.hh]: will fix, the fields will be non-configurable.

File: common/engine_legacy_conf/service.cc (hunk 5 above)
Comment (sechkem) [reply common/engine_legacy_conf/service.cc]: will fix, the fields will be non-configurable.

**Review by bouda1 — APPROVED** [review]

**Review by jean-christophe81 — APPROVED** [review]

**Review by bouda1 — APPROVED** [review]

**Pull request merged**